src/server/services/article.service.ts

```diff
@@ -302,6 +309,22 @@ export const upsertArticle = async ({
     });
     if (!article) throw throwNotFoundError(`No article with id ${id}`);
 
+    if (article && !article.publishedAt) {
+      // If it was unpublished, need to remove it from the queue.
+      if (!article.publishedAt) {
+        await articlesSearchIndex.queueUpdate([
+          { id, action: SearchIndexUpdateQueueAction.Delete },
+        ]);
+      }
```

Comment:
Feels like this `if` statement is redundant
```suggestion
      await articlesSearchIndex.queueUpdate([
        { id, action: SearchIndexUpdateQueueAction.Delete },
      ]);
```